pkg/vm/vm.go

```diff
@@ -540,7 +558,7 @@ func (vm *VirtualMachine) RelocateMemory() []*f.Element {
 	// returned has nil as its first element.
 	relocatedMemory := make([]*f.Element, maxMemoryUsed)
 	for i, segment := range vm.Memory.Segments {
-		for j := uint64(0); j < segment.Len(); j++ {
+		for j := uint64(0); j < segment.RealLen(); j++ {
```

Comment:
what is the shade here? why realLen using `Data` field instead of len using `lastIndex`

Reply:
bc lastIndex is now used for keeping the segment size after finalization, and can be greater than te size of actual data. Often the finalized builtin segment has no data in it (real len is 0), but the expected size after finalization can be for instance 2<<10. Thats why I use the `.Len()` in the generation of relocation offsets, and `.RealLen()` in accessing the actual data to be relocated

Reply:
yeah its clear thanks!